src/jsonrpc.js

```diff
@@ -17,14 +33,85 @@ export class JsonRpc {
   }
 
   /**
-   * Scanning all the blocks fromBlock and toBlock 
-   * which are internal variables passed via the constructor
+   * 
+   * This will return the final data structure 
+   * for the transaction resopnse
+   * @param string tx 
+   * @param Object receipt
+   * @param Array logs
+   * @return object
+   */
+  static getTransactionFormat(transaction, receipt, logs) {
+    const receiptResult = receipt;
+
+    receiptResult.logs = logs;
+    return { ...transaction, ...receiptResult };
+  }
+
+  /**
+   * Async function that gets the transaction and transaction receipt 
+   * then get the logs out of the receipt transaction then execute the callback function  
+   * @param string txn
+   */
+  async _scanTransaction(txn) {
+    const txnReceipts = await toPromise(this.web3Instance.getTransactionReceipt)(txn.hash);
+    const logs = txnReceipts.logs.filter(log => isInArray(this.addresses, log.address));
+
+    /**
+     * This case should never happend, because we are cheching 
```

Comment:
This case should never happen because we are checking the smart contract not the normal addresses

I would remove this comment actually as it doesn't add much

Reply:
👍